### ISSUE TITLE
Make `datalad-annex::` compatible with `datalad-osf` repo deposits

### ISSUE THREAD

**Issue by mih** [datalad/datalad-next#410]
The `git-remote-osf` helper of https://github.com/datalad/datalad-osf is the director predecessor of `datalad-annex::`. It suffers from a number of limitations that have since been improved in `datalad-annex::`. Therefore it would be nice to be able to use the latter instead of the former without loosing data access.

One critical difference is that `git-remote-osf` deposits repo archive that contain a `repo/` root directory that is absent in those deposited by `datalad-annex::`. The following patch would add legacy support for such deposits (on read only) in `datalad-annex::`

```diff
diff --git a/datalad_next/gitremotes/datalad_annex.py b/datalad_next/gitremotes/datalad_annex.py
index 90efb42..998d626 100755
--- a/datalad_next/gitremotes/datalad_annex.py
+++ b/datalad_next/gitremotes/datalad_annex.py
@@ -890,15 +890,27 @@ class RepoAnnexGitRemote(object):
             self._mirrorrepo = None
 
         self.log('Extracting repository archive')
+        legacy_deposit = False
         with zipfile.ZipFile(repoexportkeyloc) as zip:
+            try:
+                zip.getinfo('repo/')
+                legacy_deposit = True
+                safe_content = [f'repo/{i}' for i in self.safe_content]
+            except KeyError:
+                safe_content = self.safe_content
+
             zip.extractall(
                 self._mirrorrepodir,
                 # a bit of a safety-net, exclude all unexpected content
                 members=[
                     m for m in zip.namelist()
                     if any(m.startswith(prefix)
-                           for prefix in self.safe_content)],
+                           for prefix in safe_content)],
             )
+        if legacy_deposit:
+            legacy_basedir = self._mirrorrepodir / 'repo'
+            for p in legacy_basedir.iterdir():
+                p.rename(self._mirrorrepodir / p.relative_to(legacy_basedir))
 
     def get_remote_refs(self):
         """Report remote refs
```

Together with https://github.com/datalad/datalad-osf/issues/174 this would enable the following usage pattern:

```sh
git clone 'datalad-annex::?type=external&externaltype=osf&node=x3w4t&encryption=none'
```

and safe around 500 lines of code in datalad-osf.

**Comment by mih** [comment]
I was briefly considering to make the root directory in the archive configurable (both for extraction and for assembly). However, that we make users have to worry about getting it right, and also introducing flexibility that we need to account for in there future. I decided against it, until a concrete use case convinces me otherwise.

**Comment by mih** [comment]
Fixed with #411
